### ISSUE TITLE
Can't delete tags with "#" (number sign) on it

### ISSUE THREAD

**Issue by AnimMouse** [dev-drprasad/delete-older-releases#10]
I can't delete tags using this, I got this error:

```
Run dev-drprasad/delete-older-releases@v0.2.0
  with:
    keep_latest: 1
    delete_tags: true
    delete_tag_pattern: a
  env:
    GITHUB_TOKEN: ***
💬  no `repo` name given. fall-ing back to this repo
🔖  corresponding tags also will be deleted
releases containing a will be targeted
💬  found total of 2 matching active release(s)
🍻  found 1 older release(s)
starting to delete a#8 with id 39024378
🌶  failed to delete tag "a#8"  <- Reference does not exist
```

**Comment by dev-drprasad** [comment]
Message implying that there is not tag exists with name "a#8". Can you cross whether that tag exists ? Ping me repo url if it is public, I will also check

**Comment by AnimMouse** [comment]
https://github.com/AnimMouse/ffmpeg-autobuild-gpl/releases/tag/a%236

**Comment by AnimMouse** [comment]
https://github.com/AnimMouse/ffmpeg-autobuild-gpl/releases/tag/a%238

**Comment by AnimMouse** [comment]
The release was deleted, but the tag is not. Tested again on https://github.com/AnimMouse/ffmpeg-autobuild-gpl/runs/2260494888

**Comment by dev-drprasad** [comment]
If you don't mind can you release tag without "#" in it and delete it using this action. I just want to make sure "#" in tag is not the problem

**Comment by AnimMouse** [comment]
Tried tags without "#" and it works, it seems like the "#" is the problem.
```
Run dev-drprasad/delete-older-releases@v0.2.0
💬  no `repo` name given. fall-ing back to this repo
🔖  corresponding tags also will be deleted
releases containing a will be targeted
💬  found total of 3 matching active release(s)
🍻  found 2 older release(s)
starting to delete a2 with id 40909959
starting to delete a1 with id 40909957
👍🏼  2 older release(s) deleted successfully!
```

**Comment by thadguidry** [comment]
@AnimMouse we have a potential fix for this in #11 which would change to using regex for the `delete_tag_pattern` option.  Do you think that might help your situation?  Do you have strong feelings about [my comment](https://github.com/dev-drprasad/delete-older-releases/pull/11#pullrequestreview-1212559002) in that PR one way or another?

**Comment by AnimMouse** [comment]
Yes, using regex is much better.

**Comment by yoomlam** [comment]
+1 on using regex

**Comment by thadguidry** [comment]
Fixed in PR #30 and new release https://github.com/dev-drprasad/delete-older-releases/releases/tag/v0.3.1
